pyniva/request_dataframe.py

```diff
@@ -109,3 +111,136 @@ def get_ship_data(
         for path in empty_paths:
             df[path] = None
     return df
+
+def get_ramses_time_slice(              
+            vessel_name: str, 
+            param_paths: list,
+            spectral_paths: list, 
+            vessel_paths: list,
+            vessel_signals: list,
+            start_time,
+            end_time,
+            noqc,
+            header,
+            dt=0,
+            pub_tsb=PUB_TSB):
+    
+    # get spectral data
+    # first get spectral data
+    print(f"Downloading spectral data {spectral_paths[0]} times {start_time} to {end_time}")
+    df = pd.DataFrame()
+    tseries_idx = vessel_paths.index(spectral_paths[0])
+    df = vessel_signals[tseries_idx].get_tseries(
+                    pub_tsb,
+                    header=header,
+                    noqc=noqc,
+                    dt=dt,
+                    start_time=start_time,
+                    end_time=end_time)
+    if df.empty:
+        print(f"No ramses data found for path {spectral_paths[0]}")
+        return df
+    df = df.sort_values(by="time", ascending=True)
+    df = df.reset_index()
+    df = df.pivot(index = "time", columns="wl", values="value") 
+
+    # then get gps data
+    print(f"Downloading gps track")
+    path = f"{vessel_name}/gpstrack"
+    gps_index = vessel_paths.index(path)
+    gps_data = vessel_signals[gps_index].get_tseries(
+                    pub_tsb,
+                    header=header,
+                    noqc=noqc,
+                    dt=dt,
+                    start_time=start_time,
+                    end_time=end_time,
+    )
+
+    # inperpolate gps data to match ramses
+    print(f"Interpolating gps track to ramses frequency")
+    gps_data = gps_data.sort_values(by="time", ascending=True)
+    gps_data = gps_data.reset_index()
+    datetime_ramses = list(set([time.mktime(datetime.strptime(str(v), "%Y-%m-%d %H:%M:%S").timetuple()) for v in list(df.index)]))
+    gps_datetime = [time.mktime(datetime.strptime(str(v), "%Y-%m-%d %H:%M:%S").timetuple()) for v in list(gps_data["time"])]
+    longitude = list(gps_data["longitude"])
+    latitude = list(gps_data["latitude"])
+    longitude_interp = np.interp(datetime_ramses, gps_datetime, longitude)
+    latitude_interp = np.interp(datetime_ramses, gps_datetime, latitude)
+    df["longitude"] = longitude_interp
+    df["latitude"] = latitude_interp
+
+    # add other ramses data
+    for path in param_paths:
+        if path in spectral_paths:
+                continue
+        if "RAMSES" not in path:
```

Comment:
why can't regular paths be downloaded together with ramses? get_tseris should not have a problem with it

Reply:
okey...I can add it , but then I need to interpolate it too...
Do you think it would be nice to have?

Reply:
Do you think it is necessary? Will they not focus on ramses @lisapro ? It requires a bit of work, because you need to carefully consider special cases for interpolations: flags should result in 0/1  and course (angles) should include wrap around.

Reply:
In principle I see some cases it would be needed (like comparing chlorophyll or cdom with light conditions) . But if this requires work with interpolation, then maybe better to let researchers do it how they want it